torch/cuda/__init__.py

```diff
@@ -1137,6 +1137,41 @@ def _dtype(self):
 torch._storage_classes.add(ComplexDoubleStorage)
 torch._storage_classes.add(ComplexFloatStorage)
 
+
+class _WrappedTritonKernel(object):
+    """ Just a simple wrapper to store some metadata for testing purposes.
+    """
+
+    def __init__(self, kernel):
+        self.kernel = kernel
+        self.kernel_invoked = False
+
+    def __call__(self, *args, **kwargs):
+        res = self.kernel(*args, **kwargs)
+        self.kernel_invoked = True
+        return res
+
+
+def _register_triton_kernels():
+    import sys
+
+    if sys.executable == "torch_deploy":
+        return
+
+    from torch.sparse._triton_ops import bsr_dense_mm
+
+    if bsr_dense_mm is not None:
+        torch._TritonLibrary.probablyRegisterOp(
+            "_triton_bsr_dense_mm_out",
+            "_triton_bsr_dense_mm_out(Tensor bsr, Tensor dense, *, Tensor(a!) out) -> Tensor(a!)",
+            _WrappedTritonKernel(lambda *args, **kwargs: bsr_dense_mm(*args, skip_checks=True, **kwargs)),
+            "SparseCsrCUDA"
+        )
```

Comment:
the kernel wrapped into `triton.jit` is being compiled upon import plus the import, yes. So we decided to wrap all these things into a lazy call that is being cached.

Reply:
@nikitaved `triton.jit` does not compile anything! Only when you launch the kernel with a specific set of types and constexpr parameters will it be compiled into an actual kernel.

> which I guess, might not be an issue here because of Inductor?

No! Inductor very carefully avoids import triton eagerly for exactly this reason. See #100374

Reply:
cc @albanD

Reply:
Sure, @peterbell10 , thanks for the suggestion and the links. I will try this approach.

Reply:
Tried, seems to work, going with it :) Thanks, @peterbell10 !